model/lifecycle.go

```diff
@@ -1703,14 +1703,24 @@ func addNewTasks(ctx context.Context, creationInfo TaskCreationInfo, existingBui
 		// a record in the TVPairSet indicating that it should exist
 		tasksToAdd := []string{}
 		for _, taskName := range creationInfo.Pairs.ExecTasks.TaskNames(b.BuildVariant) {
-			if ok := existingTasksIndex[taskName]; ok {
+			if t, ok := existingTasksIndex[taskName]; ok {
+				if !t.Activated {
+					if err = SetActiveState(evergreen.DefaultTaskActivator, true, *t); err != nil {
+						return nil, errors.Wrapf(err, "setting task '%s' to active", t.Id)
```

Comment:
nit: can we mention in the error wrapping message that we're activating an execution task rather than a normal task here?